ql/src/java/org/apache/hadoop/hive/ql/udf/UDFSubstr.java

```diff
@@ -85,6 +107,31 @@ public Text evaluate(Text t, IntWritable pos, IntWritable len) {
     return r;
   }
 
+  private int[] makeIndex(long pos, long len, int inputLen) {
+    if ((Math.abs(pos) > inputLen)) {
+      return null;
+    }
+
+    long start, end;
+
+    if (pos > 0) {
+      start = pos - 1;
+    } else if (pos < 0) {
+      start = inputLen + pos;
+    } else {
+      start = 0;
+    }
+
+    if ((inputLen - start) < len) {
+      end = inputLen;
+    } else {
+      end = start + len;
+    }
+    index[0] = (int) start;
+    index[1] = (int) end;
+    return index;
+  }
```

Comment:
This is unnecessary if both `LongWritable` and `IntWritable` `evaluate` call the common  one. See my previous comment.

Reply:
Done